app/src/main/java/com/amaze/filemanager/ui/fragments/MainFragment.java

```diff
@@ -1826,4 +1817,45 @@ public int getRootDrawable() {
   private boolean getBoolean(String key) {
     return getMainActivity().getBoolean(key);
   }
+
+  @Override
+  public void onGlobalLayout() {
+    if (mainFragmentViewModel.getColumns() == 0 || mainFragmentViewModel.getColumns() == -1) {
+      int screen_width = listView.getWidth();
+      int dptopx = Utils.dpToPx(getContext(), 115);
+      mainFragmentViewModel.setColumns(screen_width / dptopx);
+      if (mainFragmentViewModel.getColumns() == 0 || mainFragmentViewModel.getColumns() == -1) {
+        mainFragmentViewModel.setColumns(3);
+      }
+      if (!mainFragmentViewModel.isList()) {
+        mLayoutManagerGrid.setSpanCount(mainFragmentViewModel.getColumns());
+      }
+    }
+    if (!mainFragmentViewModel.isList()) {
+      loadViews();
+    }
+    if (android.os.Build.VERSION.SDK_INT >= JELLY_BEAN) {
+      mToolbarContainer.getViewTreeObserver().removeOnGlobalLayoutListener(this);
+    } else {
+      mToolbarContainer.getViewTreeObserver().removeGlobalOnLayoutListener(this);
+    }
+  }
+
+  @Override
+  public boolean onTouch(View v, MotionEvent event) {
+    switch (v.getId()) {
+      case R.id.listView:
+      case R.id.lin:
+        if (adapter != null && mainFragmentViewModel.getStopAnims()) {
+          stopAnimation();
+          mainFragmentViewModel.setStopAnims(false);
+        }
+        return false;
+    }
+    return true;
+  }
+
+  public MainFragmentViewModel getMainFragmentViewModel() {
+    return this.mainFragmentViewModel;
```

Comment:
This is wrong, this should get an instance of the lifecycle independent holder of the model, and get an instance of the model from there.